### ISSUE TITLE
Improve Accumulator implementation, including proof verify and store

### ISSUE THREAD

**Issue by baichuan3** [rooch-network/rooch#412]
Currently, only the inmemory version of the Accumulator has been implemented. Next, consider improving the storage and proof verify features of the Accumulator.

**Comment by bytedeveloperr** [comment]
I will try working on this

**Comment by bytedeveloperr** [comment]
@jolestar @baichuan3 should the `InMemoryAccumulator` struct still remain? so we'll have another struct called `Accumulator` or something.

**Comment by baichuan3** [comment]
> @jolestar @baichuan3 should the `InMemoryAccumulator` struct still remain? so we'll have another struct called `Accumulator` or something.
> 
> or just remove the `InMemoryAccumulator` struct entirely?

yes, `InMemoryAccumulator` should remain  for test scenarios,  you can have another struct called `Accumulator` or `AccumulatorInfo`

**Comment by jolestar** [comment]
InMemoryAccumulator is also for generating Merkle tree root for events.
